app/Http/Controllers/Api/RoleController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\RolePermissionRequest;
+use App\Http\Requests\Api\RoleRequest;
+use Illuminate\Http\Response;
+use Spatie\Permission\Models\Role;
+
+class RoleController extends Controller
+{
+    public function index(): Response
+    {
+        $roles = Role::all();
+
+        return response($roles);
+    }
+
+    public function store(RoleRequest $request): Response
+    {
+        $validated = $request->validated();
+        $role = Role::create($validated);
+
+        return response($role);
+    }
+
+    public function show(Role $role): Response
+    {
+        return response($role);
+    }
+
+    public function update(RoleRequest $request, Role $role): Response
+    {
+        $validated = $request->validated();
+        $role->update($validated);
+
+        return response($role);
+    }
+
+    public function destroy(Role $role): Response
+    {
+        $role->delete();
+
+        return response()->noContent();
+    }
+
+    public function updateRolePermissions(RolePermissionRequest $request, Role $role): Response
+    {
+        $validated = $request->validated();
+        $permissions = $validated['permissions'];
+        $roleWithUpdatedPermission = $role->syncPermissions($permissions);
+
+        return response($roleWithUpdatedPermission);
```

Comment:
> I think we should return the role with new set of permissions as a response from this API.

yes, it's already returned the role with the new set of permissions as a response, I think the name `isUpdated` suggests that it is returning a boolean value that's why I changed the variable name according to their return statement, `$roleWithUpdatedPermission` is return the role with the new set of permissions.